core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSail.java

```diff
@@ -967,10 +967,61 @@ public IsolationLevel getDefaultIsolationLevel() {
 
 	public static class TransactionSettings {
 
+		public enum PerformanceHint implements TransactionSetting {
+
+			/**
+			 * Run validation is paralle (multithreaded).
+			 */
+			ParallelValidation("ParallelValidation"),
+			/**
+			 * Run validation serially (single threaded)
+			 */
+			SerialValidation("SerialValidation"),
```

Comment:
These are mutually exclusive. What happens if you supply both?

Reply:
Since the user effectively provides an array of transaction settings, the first ones will be overridden be the next ones.